File: Selection/selection.go
package main

import "fmt"

func main() {

	// initional an arr with auto length
	arr := [...]int{2,5,8,20,6,10,13,21,43,15}

	// for loop to get each element
	for i := 0; i < len(arr)-1; i++ {

		// initional the index of minimun element
		minPosition := i;

		// compare the element to get the minimum index
		for j:=i+1; j< len(arr); j++ {

			// go can't use the conditional operator
			if(arr[j] < arr[minPosition]) {
				minPosition = j
			}
		}

		// exchange the element
		tmp := arr[i]
		arr[i] = arr[minPosition]
		arr[minPosition] = tmp
	}
	fmt.Println(arr)
}

File: Selection/selection.js
// Time complexity o(n2), unstable

// initional an array
var arr = [10, 2, 3, 6, 5, 9, 15, 14];

// using for loop to interate through the array
for (var i = 0; i < arr.length - 1; i++){
  
  // get the index of the minimum element
  var minPosition = i;
  // using for loop to interate through the array 
  for (var j = i + 1; j < arr.length; j++) {

    // using conditional operator to get the minimum element
    minPosition = arr[j] < arr[minPosition] ? j : minPosition;
  }

  // exchange the elements
  getExchange(arr, i, minPosition);

  // Print the result of each element change
  console.log("this is the step " + i + " The result is " + arr);
}

// function to exchange two elements position
function getExchange (arr, i, j) {
  var tmp = arr[i];
  arr[i] = arr[j];
  arr[j] = tmp;
}

